File: backend/src/main/scala/merchant/object/menu/AddMenuItemRequest.scala
package domain.merchant

import domain.shared.given

import domain.shared.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class AddMenuItemRequest(
    name: DisplayText,
    category: Option[DisplayText],
    description: DescriptionText,
    priceCents: CurrencyCents,
    imageUrl: Option[ImageUrl],
    remainingQuantity: Option[Quantity],
    selectionGroups: List[MenuItemSelectionGroup],
)
object AddMenuItemRequest:
  given Encoder[AddMenuItemRequest] = deriveEncoder
  given Decoder[AddMenuItemRequest] = deriveDecoder

File: frontend/src/merchant/object/menu/AddMenuItemRequest.ts
import type {
  CurrencyCents,
  DescriptionText,
  DisplayText,
  ImageUrl,
  Quantity,
} from '@/shared/object/domain/DomainObjects'
import type { MenuItemSelectionGroup } from '@/merchant/object/menu/MenuItemSelectionGroup'

export type AddMenuItemRequest = {
  name: DisplayText
  category?: DisplayText
  description: DescriptionText
  priceCents: CurrencyCents
  imageUrl?: ImageUrl
  remainingQuantity?: Quantity
  selectionGroups: MenuItemSelectionGroup[]
}
